mkosi/kmod.py

```diff
@@ -18,41 +20,101 @@
 def loaded_modules() -> list[str]:
     # Loaded modules are listed with underscores but the filenames might use dashes instead.
     return [
-        rf"/{line.split()[0].replace('_', '[_-]')}\.ko"
-        for line in Path("/proc/modules").read_text().splitlines()
+        normalize_module_name(line.split()[0]) for line in Path("/proc/modules").read_text().splitlines()
     ]
 
 
+def globs_match_filename(name, globs) -> Optional[bool]:
```

Comment:
The `Optional[bool]` return makes this a bit hard to understand, since one has to look at the users of the function to understand the difference between `False` and `None`. How about this instead
```suggestion
def globs_match_filename(name, globs, *, match_default: bool: False) -> bool:
```
and return `match_default` at the end. The option would need to be added to `glob_match_firmware`, but it would make this a simple yes/no predicate again and sidestep the question "Why is there an `is False` check there?" in the firmware inclusion code.